### PR TITLE
Remove injecting observability checkPointCall for desugared code

### LINKED ISSUE #37694: [Bug]: GraalVM native image generation failed with retry transaction block
### Description

Sample code

```ballerina
public function main() returns error? {
    retry transaction {
        check commit;
    }
}
```

### Steps to Reproduce

_No response_

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Labels

jBallerina

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject

Fixes #37694 

## Approach
The issue seemingly happens due to multiple injections of the observability method call for `recordCheckpoint()`. 
This PR prevents adding additional observability calls for the desugared code for the `retry-transaction` clause by passing `null` as the location. 

## Samples
```ballerina
public function main() returns error? {
    retry transaction {
        check commit;
    }
}
```
The graalVM native build hangs with an OOM error.

## CheckList 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by Nadeeshan96 — COMMENTED** [review]
Have we tested this fix by running the tests in `module-ballerina-observe` and `module-ballerinai-observe` repos? If not shall we do it?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 1]
```diff
@@ -3341,15 +3341,21 @@ protected BLangTypeTestExpr createTypeCheckExpr(Location pos, BLangExpression ex
...
         BLangSimpleVariableDef prevAttemptVarDef = transactionDesugar.createPrevAttemptInfoVarDef(env,
                 retryTransaction.pos);
         retryTransaction.transaction.prevAttemptInfo = ASTBuilderUtil.createVariableRef(retryTransaction.pos,
                 prevAttemptVarDef.var.symbol);
 
+        // the positions are removed to prevent unnecessary observability calls from desugared code.
+        retryTransaction.pos = null;
+        retryTransaction.retrySpec.pos = null;
+        retryTransaction.transaction.pos = null;
+        retryTransaction.transaction.transactionBody.pos = null;
```
Comment (gabilang) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:3354]: Won't removing positions affect the debugger behavior?

Comment (HindujaB) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:3354]: The behavior looks the same after the change when I checked it locally.
Screenshot from 2022-11-10 08-29-46

**Comment by HindujaB** [comment]
> Have we tested this fix by running the tests in `module-ballerina-observe` and `module-ballerinai-observe` repos? If not shall we do it?

The earlier fix was breaking observability tracing for some instructions (Eg: `foreach` loop and  inline object creation). 
I modified the fix to be specific to retry-transaction blocks. 
Tested and verified the following. 
- graalVM native build with 12 iterations
- `module-ballerinai-transaction` build with tests
-  `module-ballerinai-observe` build with tests
-  `module-ballerina-observe` build with tests

**Review by gabilang — APPROVED** [review]
LGTM

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Pull request merged**